This patch makes it possible to install to a directory tree rooted
elsewhere by saying "make install DESTDIR=$PWD/debian/tmp".  As you
can guess, this is useful for building Debian packages.


To apply this in the gettext sources:

# cd /usr/local/src/gettext-0.10.35
(or wherever you have extracted the source tree)
# patch -p1 </PATH/TO/gettext-0.10.35-5-destdir.diff

After installing the patched gettext, rerun `gettextize -f' in your
favorite packages to let them use the changes.


I (Kalle Niemitalo) don't think copyright applies to changes as simple
as these.


diff -Nurp gettext-0.10.35.kn.4/po/ChangeLog gettext-0.10.35.kn.5/po/ChangeLog
--- gettext-0.10.35.kn.4/po/ChangeLog	Wed Apr 29 04:07:27 1998
+++ gettext-0.10.35.kn.5/po/ChangeLog	Fri Dec 18 17:53:20 1998
@@ -1,3 +1,9 @@
+1998-11-05  Kalle Olavi Niemitalo  <tosi@stekt.oulu.fi>
+
+	* Makefile.in.in (DESTDIR): New variable.  It's empty unless
+	overridden in command line.  If specified, it's the root directory
+	for installation.  This makes it easier to create Debian packages.
+
 1998-04-28 18:07  Ulrich Drepper  <drepper@cygnus.com>
 
 	* po/Makefile.in.in: Only try to put newly generated .pot in place
diff -Nurp gettext-0.10.35.kn.4/po/Makefile.in.in gettext-0.10.35.kn.5/po/Makefile.in.in
--- gettext-0.10.35.kn.4/po/Makefile.in.in	Wed Apr 29 04:53:09 1998
+++ gettext-0.10.35.kn.5/po/Makefile.in.in	Fri Dec 18 17:11:47 1998
@@ -24,6 +24,8 @@ gnulocaledir = $(prefix)/share/locale
 gettextsrcdir = $(prefix)/share/gettext/po
 subdir = po
 
+DESTDIR =
+
 INSTALL = @INSTALL@
 INSTALL_DATA = @INSTALL_DATA@
 MKINSTALLDIRS = $(top_srcdir)/@MKINSTALLDIRS@
@@ -111,9 +113,9 @@ install-data: install-data-@USE_NLS@
 install-data-no: all
 install-data-yes: all
 	if test -r "$(MKINSTALLDIRS)"; then \
-	  $(MKINSTALLDIRS) $(datadir); \
+	  $(MKINSTALLDIRS) $(DESTDIR)$(datadir); \
 	else \
-	  $(SHELL) $(top_srcdir)/mkinstalldirs $(datadir); \
+	  $(SHELL) $(top_srcdir)/mkinstalldirs $(DESTDIR)$(datadir); \
 	fi
 	@catalogs='$(CATALOGS)'; \
 	for cat in $$catalogs; do \
@@ -123,7 +125,7 @@ install-data-yes: all
 	    *)     destdir=$(localedir);; \
 	  esac; \
 	  lang=`echo $$cat | sed 's/\$(CATOBJEXT)$$//'`; \
-	  dir=$$destdir/$$lang/LC_MESSAGES; \
+	  dir=$(DESTDIR)$$destdir/$$lang/LC_MESSAGES; \
 	  if test -r "$(MKINSTALLDIRS)"; then \
 	    $(MKINSTALLDIRS) $$dir; \
 	  else \
@@ -144,7 +146,7 @@ install-data-yes: all
 	    if test -r $(srcdir)/$$cat.m ; then \
 	      $(INSTALL_DATA) $(srcdir)/$$cat.m \
 		$$dir/$(PACKAGE)$(INSTOBJEXT).m; \
-	      echo "installing $(srcdir)/$$cat as" \
+	      echo "installing $(srcdir)/$$cat.m as" \
 		   "$$dir/$(PACKAGE)$(INSTOBJEXT).m"; \
 	    else \
 	      true; \
@@ -153,12 +155,12 @@ install-data-yes: all
 	done
 	if test "$(PACKAGE)" = "gettext"; then \
 	  if test -r "$(MKINSTALLDIRS)"; then \
-	    $(MKINSTALLDIRS) $(gettextsrcdir); \
+	    $(MKINSTALLDIRS) $(DESTDIR)$(gettextsrcdir); \
 	  else \
-	    $(SHELL) $(top_srcdir)/mkinstalldirs $(gettextsrcdir); \
+	    $(SHELL) $(top_srcdir)/mkinstalldirs $(DESTDIR)$(gettextsrcdir); \
 	  fi; \
 	  $(INSTALL_DATA) $(srcdir)/Makefile.in.in \
-			  $(gettextsrcdir)/Makefile.in.in; \
+			  $(DESTDIR)$(gettextsrcdir)/Makefile.in.in; \
 	else \
 	  : ; \
 	fi
@@ -171,12 +173,12 @@ uninstall:
 	for cat in $$catalogs; do \
 	  cat=`basename $$cat`; \
 	  lang=`echo $$cat | sed 's/\$(CATOBJEXT)$$//'`; \
-	  rm -f $(localedir)/$$lang/LC_MESSAGES/$(PACKAGE)$(INSTOBJEXT); \
-	  rm -f $(localedir)/$$lang/LC_MESSAGES/$(PACKAGE)$(INSTOBJEXT).m; \
-	  rm -f $(gnulocaledir)/$$lang/LC_MESSAGES/$(PACKAGE)$(INSTOBJEXT); \
-	  rm -f $(gnulocaledir)/$$lang/LC_MESSAGES/$(PACKAGE)$(INSTOBJEXT).m; \
+	  rm -f $(DESTDIR)$(localedir)/$$lang/LC_MESSAGES/$(PACKAGE)$(INSTOBJEXT); \
+	  rm -f $(DESTDIR)$(localedir)/$$lang/LC_MESSAGES/$(PACKAGE)$(INSTOBJEXT).m; \
+	  rm -f $(DESTDIR)$(gnulocaledir)/$$lang/LC_MESSAGES/$(PACKAGE)$(INSTOBJEXT); \
+	  rm -f $(DESTDIR)$(gnulocaledir)/$$lang/LC_MESSAGES/$(PACKAGE)$(INSTOBJEXT).m; \
 	done
-	rm -f $(gettextsrcdir)/po-Makefile.in.in
+	rm -f $(DESTDIR)$(gettextsrcdir)/po-Makefile.in.in
 
 check: all
 
